src/NuGet.Clients/NuGet.CommandLine/Common/ProtocolDiagnosticsJsonLogger.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using Newtonsoft.Json;
+using NuGet.Protocol.Utility;
+
+namespace NuGet.CommandLine
+{
+    internal sealed class ProtocolDiagnosticsJsonLogger : IDisposable
```

Comment:
Consider using the HAR file format as inspiration. It has rich tooling

Reply:
I didn't know about the HAR format, so thanks for letting me know. However, it adds a lot of fields that will not be relevant for file feeds/sources. I'm also not sure if logging to a file will make it into the product. I added this for testing to ensure the events are working properly, which I'll still need when I add file feed instrumentation, but it may get deleted before merging into dev.